tests/integration/embed/embed_test.go

```diff
@@ -210,3 +210,19 @@ func setupEmbedCfg(cfg *embed.Config, curls []url.URL, purls []url.URL) {
 	}
 	cfg.InitialCluster = cfg.InitialCluster[1:]
 }
+
+func TestEmbedEtcdAutoCompactionRetentionRetained(t *testing.T) {
```

Comment:
My understanding is that this test adds another use case to `TestAutoCompactionModeParse`. Where `mode == ""` and `retention == ''2"`, so I'd just update TestAutoCompactionModeParse and remove this test.
There is probably an opportunity to add integration test for compaction mode and retention,  but it shouldn't stop at verifying config.